From 17c7711948f8157478dc3abe9c52b8078fd8b5a9 Mon Sep 17 00:00:00 2001
From: Kohei Yoshida <kohei.yoshida@suse.com>
Date: Wed, 7 Dec 2011 10:38:25 -0500
Subject: [PATCH] fdo#43308: Set the logic straight for "center across
 selection" setting.

Another unfortunate logic change introduced via DECLARE_LIST removal.
Reverting the logic to the original one.
---
 sc/source/filter/excel/xistyle.cxx |   12 +++---------
 1 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/sc/source/filter/excel/xistyle.cxx b/sc/source/filter/excel/xistyle.cxx
index d2d4467..2131095 100644
--- a/sc/source/filter/excel/xistyle.cxx
+++ b/sc/source/filter/excel/xistyle.cxx
@@ -1824,15 +1824,9 @@ void XclImpXFRangeBuffer::SetXF( const ScAddress& rScPos, sal_uInt16 nXFIndex, X
         if( pXF && ((pXF->GetHorAlign() == EXC_XF_HOR_CENTER_AS) || (pXF->GetHorAlign() == EXC_XF_HOR_FILL)) )
         {
             // expand last merged range if this attribute is set repeatedly
-            if ( !maMergeList.empty() )
-            {
-                ScRange* pRange = maMergeList.back();
-                if(  (pRange->aEnd.Row()     == nScRow)
-                  && (pRange->aEnd.Col() + 1 == nScCol)
-                  && (eMode                  == xlXFModeBlank)
-                  )
-                    pRange->aEnd.IncCol();
-            }
+            ScRange* pRange = maMergeList.empty() ? NULL : maMergeList.back();
+            if (pRange && (pRange->aEnd.Row() == nScRow) && (pRange->aEnd.Col() + 1 == nScCol) && (eMode == xlXFModeBlank))
+                pRange->aEnd.IncCol();
             else if( eMode != xlXFModeBlank )   // do not merge empty cells
                 SetMerge( nScCol, nScRow );
         }
-- 
1.7.6.4

